src/Mono.Android/Android.Runtime/InputStreamInvoker.cs

```diff
@@ -58,11 +60,7 @@ protected override void Dispose (bool disposing)
 		//
 		public override void Close ()
 		{
-			try {
-				BaseInputStream.Close ();
-			} catch (Java.IO.IOException ex) when (JNIEnv.ShouldWrapJavaException (ex)) {
-				throw new IOException (ex.Message, ex);
-			}
+			base.Close ();
```

Comment:
Can `base.Close()` ever throw a `Java.IO.IOException`, or is that handled elsewhere?

Reply:
It's handled in `Dispose(bool)`: `base.Close()` is `Stream.Close()`, which is: https://github.com/dotnet/runtime/blob/2ea6ae57874c452923af059cbcb57d109564353c/src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs#L158-L165

```csharp
        public virtual void Close()
        {
            // When initially designed, Stream required that all cleanup logic went into Close(),
            // but this was thought up before IDisposable was added and never revisited. All subclasses
            // should put their cleanup now in Dispose(bool).
            Dispose(true);
            GC.SuppressFinalize(this);
        }
```

Meanwhile, our `Dispose(bool)` override *does* translate `Java.IO.IOException` to `System.IO.IOException`.